exe/kubernetes-deploy

```diff
@@ -42,27 +46,31 @@ end
 
 namespace = ARGV[0]
 context = ARGV[1]
-template_dir = KubernetesDeploy::OptionsHelper.default_and_check_template_dir(template_dir)
 logger = KubernetesDeploy::FormattedLogger.build(namespace, context, verbose_prefix: verbose_log_prefix)
 
-runner = KubernetesDeploy::DeployTask.new(
-  namespace: namespace,
-  context: context,
-  current_sha: ENV["REVISION"],
-  template_dir: template_dir,
-  bindings: bindings,
-  logger: logger,
-  max_watch_seconds: max_watch_seconds
-)
-
 begin
-  runner.run!(
-    verify_result: !skip_wait,
-    allow_protected_ns: allow_protected_ns,
-    prune: prune
-  )
+  KubernetesDeploy::OptionsHelper.with_consolidated_template_dir(template_dirs) do |template_dir|
+    runner = KubernetesDeploy::DeployTask.new(
+      namespace: namespace,
+      context: context,
+      current_sha: ENV["REVISION"],
+      template_dir: template_dir,
+      bindings: bindings,
+      logger: logger,
+      max_watch_seconds: max_watch_seconds
+    )
+
+    runner.run!(
+      verify_result: !skip_wait,
+      allow_protected_ns: allow_protected_ns,
+      prune: prune
+    )
+  end
 rescue KubernetesDeploy::DeploymentTimeoutError
   exit(70)
 rescue KubernetesDeploy::FatalDeploymentError
   exit(1)
+rescue KubernetesDeploy::OptionsHelper::OptionsError => e
+  puts e.message
```

Comment:
Do we want to do the logging here or in `with_consolidated_template_dir`. If here, should this be using `logger` instead of `puts`

Reply:
Makes sense for me to use the logger. I'll update